src/main/java/uk/gov/hmcts/ccd/endpoint/std/CaseSearchEndpoint.java

```diff
@@ -66,7 +66,12 @@ public CaseSearchResult searchCases(
 
         rejectBlackListedQuery(jsonSearchRequest);
 
-        CrossCaseTypeSearchRequest request = new CrossCaseTypeSearchRequest(caseTypeIds, convertJsonStringToJsonNode(jsonSearchRequest));
+        CrossCaseTypeSearchRequest request = new CrossCaseTypeSearchRequest.Builder()
+            .withCaseTypes(caseTypeIds)
+            .withSearchRequest(convertJsonStringToJsonNode(jsonSearchRequest))
+            .withMultiCaseTypeSearch(caseTypeIds.size() > 1)
```

Comment:
I liked more the previous approach, i.e. the CrossCaseTypeSearchRequest itself determines if it's a multi search by checking the size of the collection
Less code required

Reply:
moved